src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -89,7 +88,8 @@ public OAuthHandler(HttpClient backchannel)
                 var value = query["error"];
                 if (!StringValues.IsNullOrEmpty(value))
                 {
-                    Logger.LogVerbose("Remote server returned an error: " + Request.QueryString);
+                    Logger.LogWarning("Remote server returned an error: " + value);
```

Comment:
Nah, forget OIDC, it doesn't work AT ALL.
In this basic sample, it doesn't redirect the user:

```csharp
OnAuthenticationFailed = context => {
    if (string.Equals(context.ProtocolMessage.Error, "access_denied", StringComparison.Ordinal)) {
        context.HandleResponse();

        context.Response.Redirect("/");
    }

    return Task.FromResult<object>(null);
}
```

https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/samples/Mvc/Mvc.Client/Startup.cs#L63-L74